fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1001,10 +1017,13 @@ ACTOR Future<Void> assignMutationsToStorageServers(CommitBatchContext* self) {
 					    .detail("Mutation", m);
 
 					self->toCommit.addTags(allSources);
+
+					self->toCommit.saveTags(self->tagSet);
```

Comment:
A quick comment about this: Does "saveTags()" need to be called in each branch? Can't we call this once per mutation?

Reply:
in each branch, the initial set of tags is built differently, and that set is in turn converted to locations differently. It would take some refactoring to reconcile the different cases which is beyond this PR's scope.